### PR TITLE
fs-presence

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/callflow/src/module/cf_park.erl [hunk 1]
```diff
@@ -732,23 +722,22 @@ get_endpoint_id(Username, Call) ->
...
 %%--------------------------------------------------------------------
--spec ringback_parker(api_binary(), ne_binary(), ne_binary(), kz_json:object(), kapps_call:call()) ->
+-spec ringback_parker(api_binary(), ne_binary(), kz_json:object(), kapps_call:call()) ->
                              'answered' | 'failed' | 'channel_hungup'.
-ringback_parker('undefined', _, _, _, _) -> 'failed';
-ringback_parker(EndpointId, SlotNumber, TmpCID, Data, Call) ->
+ringback_parker('undefined', _, _, _) -> 'failed';
+ringback_parker(EndpointId, SlotNumber, Data, Call0) ->
+    TmpCID = <<"Parking slot ", SlotNumber/binary>>,
+    Call = kapps_call:kvs_store('dynamic_cid', {'undefined', TmpCID}, Call0),
```
Comment (icehess) [inline applications/callflow/src/module/cf_park.erl]: why setting cid_number to `undefined`?

**Comment by icehess** [comment]
@lazedo other than comment about caller_id_number, LGTM

File: applications/callflow/src/module/cf_park.erl (hunk 1 above)
Comment (lazedo) [inline applications/callflow/src/module/cf_park.erl]: when ringing back the parker, we want to keep the initial number but change the `display`

File: applications/ecallmgr/src/ecallmgr_fs_presence.erl [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+-spec process_specific_event(ne_binary(), api_binary(), kz_proplist(), atom()) -> any().
+process_specific_event(<<"PRESENCE_IN">>, UUID, Props, Node) ->
+    AuthType = props:get_value(?GET_CCV(<<"Authorizing-Type">>), Props),
+    maybe_build_presence_event(Node, UUID, Props, AuthType);
+process_specific_event(_Event, _UUID, _Props, _Node) ->
+    lager:debug("event ~s for callid ~s not handled in presence (~s)", [_Event, _UUID, _Node]).
+
+
+maybe_build_presence_event(Node, UUID, Props, <<"device">>) ->
```
Comment (k-anderson) [inline applications/ecallmgr/src/ecallmgr_fs_presence.erl]: spec

File: applications/callflow/src/module/cf_park.erl [hunk 3]
```diff
@@ -773,6 +762,10 @@ wait_for_ringback(Fun, Timeout, Call) ->
             'failed'
     end.
 
+expires(<<"early">>) -> 3600;
```
Comment (k-anderson) [inline applications/callflow/src/module/cf_park.erl]: spec

File: applications/ecallmgr/src/ecallmgr_fs_presence.erl [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    build_presence_event(Node, UUID, Props);
+maybe_build_presence_event(_Node, _UUID, _Props, _AuthType) ->
+    lager:debug("Authorizing-Type ~p not handled", [_AuthType]).
+
+from(Props) ->
+    props:get_first_defined([<<"from">>
+                            ,<<"variable_presence_id">>
+                            ,<<"Channel-Presence-ID">>
```
Comment (k-anderson) [inline applications/ecallmgr/src/ecallmgr_fs_presence.erl]: I think that the presence-id should be used prior to 'from'... no?

Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_fs_presence.erl]: the `from` was already set with presence_id in fs

File: core/kazoo_amqp/src/api/kapi_presence.erl [hunk 5]
```diff
@@ -171,6 +189,23 @@
                      ]).
 -define(SYNC_TYPES, []).
 
+%% Dialog state updates
+-define(DIALOG_HEADERS, [<<"To">>, <<"From">>]).
+-define(OPTIONAL_DIALOG_HEADERS, [<<"Call-ID">>, <<"Direction">>
```
Comment (k-anderson) [inline core/kazoo_amqp/src/api/kapi_presence.erl]: in ecallmgr_fs_presence the payload is built with Presence-ID but I don't see it in the list.  I might just be overlooking it...

File: core/kazoo_amqp/src/api/kapi_presence.erl [hunk 6]
```diff
@@ -450,7 +575,6 @@ publish_mwi_query(Req, ContentType) ->
     {'ok', Payload} = kz_api:prepare_api_payload(Req, ?MWI_QUERY_VALUES, fun mwi_query/1),
     amqp_util:presence_publish(mwi_query_routing_key(Req), Payload, ContentType).
 
-%% <<"Notify-User">>, <<"Notify-Realm">>
 mwi_query_routing_key(Prop) when is_list(Prop) ->
```
Comment (k-anderson) [inline core/kazoo_amqp/src/api/kapi_presence.erl]: spec

**Pull request merged**